openquake/baselib/node.py

```diff
@@ -644,10 +654,16 @@ def node_to_dict(node):
     if isinstance(node.text, str) and node.text.strip() == '':
         pass
     elif node.text is not None:
-        if 'attrib' in dic:
-            dic['text'] = node.text
+        if node.attrib:
+            try:
+                dic['text'] = float(node.text)
+            except (TypeError, ValueError):
+                dic['text'] = node.text
         else:
-            dic = node.text
+            try:
+                dic = float(node.text)
+            except (TypeError, ValueError):
+                dic = node.text
```

Comment:
This is not the way to do the conversion, for instance it would silently convert int intto floats and we do not want that. Let's talk.